File: Search_Tech_Games_III/dp.cpp
#include<iostream>
using namespace std;

int main ()
{
  int f[105], i, cur_money, cost;

  cur_money = 15;
  f[0] = 0;

  for (i = 1; i <= cur_money; i++)
    {
      cost = 99999;
      if (i - 1 >= 0) cost = min (cost, f[i - 1] + 1);
      if (i - 5 >= 0) cost = min (cost, f[i - 5] + 1);
      if (i - 11 >= 0) cost = min (cost, f[i - 11] + 1);
      f[i] = cost;
      printf ("f[%d]=%d\n", i, f[i]);
    }

  return 0;
}

File: Search_Tech_Games_III/dp.py

def main():

    f = []
    cur_money = 15
    f.append(0)  # f[0] == 0

    for i in range(1, cur_money + 1):
        cost = 99999
        if i - 1 >= 0:
            cost = min(cost, f[i-1] + 1)

        if i - 5 >= 0:
            cost = min(cost, f[i-5] + 1)

        if i - 11 >= 0:
            cost = min(cost, f[i-11] + 1)

        f.append(cost)
        print("f[%d]=%d" % (i, f[i]))


if __name__ == "__main__":
    main()
